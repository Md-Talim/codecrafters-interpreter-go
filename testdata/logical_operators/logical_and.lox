// Test Case 1
if (false and "bad") print "foo";
if (nil and "bad") print "foo";

if (true and "hello") print "hello";
if (97 and "baz") print "baz";
if ("baz" and "baz") print "baz";
if ("" and "bar") print "bar";

// Test Case 2
print false and 1;
print true and 1;
print 23 and "hello" and false;

print 23 and true;
print 23 and "hello" and 23;

// Test Case 3
var a = "quz";
var b = "quz";
(a = true) and (b = false) and (a = "bad");
print a;
print b;