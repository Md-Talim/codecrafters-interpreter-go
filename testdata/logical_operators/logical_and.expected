hello
baz
baz
bar
false
1
false
true
23
true
false